### PR TITLE
cli/wallet: show tx fee before relaying

### LINKED ISSUE #2175: Show/confirm GAS spent for fees in CLI
See neo-project/neo-node#823. Some "force" flag is also needed and this should be done for all transactions created.

### PR DESCRIPTION
**Pull request by fyrchik**
Fix #2175 .
Signed-off-by: Evgeniy Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: cli/wallet/nep17.go [hunk 1]
```diff
@@ -580,6 +582,12 @@ func signAndSendNEP17Transfer(ctx *cli.Context, c *client.Client, acc *wallet.Ac
 			return cli.NewExitError(err, 1)
 		}
 	} else {
+		if !ctx.Bool("force") {
```
Comment (AnnaShaleva) [inline cli/wallet/nep17.go:585]: Don't we want to ask confirmation for smartcontract's `deploy` and `invokefunction` and for validator commands?

File: cli/wallet/wallet.go [hunk 2]
```diff
@@ -734,3 +735,17 @@ func fmtPrintWallet(w io.Writer, wall *wallet.Wallet) {
 	fmt.Fprintln(w, string(b))
 	fmt.Fprintln(w, "")
 }
+
+// ConfirmTx asks for a confirmation to send tx.
+func ConfirmTx(w io.Writer, tx *transaction.Transaction) error {
+	fmt.Fprintf(w, "NetworkFee: %d\n", tx.NetworkFee)
+	fmt.Fprintf(w, "TotalFee: %d\n", tx.NetworkFee+tx.SystemFee)
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:742]: Better output system fee and then `Total`.

File: cli/wallet/wallet.go [hunk 3]
```diff
@@ -734,3 +735,17 @@ func fmtPrintWallet(w io.Writer, wall *wallet.Wallet) {
 	fmt.Fprintln(w, string(b))
 	fmt.Fprintln(w, "")
 }
+
+// ConfirmTx asks for a confirmation to send tx.
+func ConfirmTx(w io.Writer, tx *transaction.Transaction) error {
+	fmt.Fprintf(w, "NetworkFee: %d\n", tx.NetworkFee)
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:741]: `Network fee`, it's for humans.

File: cli/wallet/wallet.go [hunk 4]
```diff
@@ -734,3 +735,17 @@ func fmtPrintWallet(w io.Writer, wall *wallet.Wallet) {
...
+	fmt.Fprintf(w, "TotalFee: %d\n", tx.NetworkFee+tx.SystemFee)
+	ln, err := input.ReadLine("Relay tx (y|N)> ")
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:743]: s/tx/transaction/

**Review by roman-khimov — COMMENTED** [review]
We also need it for contract invocations (where we have `--force` already and it's OK to make this flag change fee behavior).

File: cli/wallet/nep17.go [hunk 5]
```diff
@@ -75,6 +77,7 @@ func newNEP17Commands() []cli.Command {
 	balanceFlags = append(balanceFlags, options.RPC...)
 	transferFlags := make([]cli.Flag, len(baseTransferFlags))
 	copy(transferFlags, baseTransferFlags)
+	transferFlags = append(transferFlags, forceFlag)
```
Comment (AnnaShaleva) [inline cli/wallet/nep17.go:80]: Can't you add it to `baseTransferFlags`?

File: cli/candidate_test.go [hunk 6]
```diff
@@ -22,6 +22,7 @@ func TestRegisterCandidate(t *testing.T) {
 		"--rpc-endpoint", "http://"+e.RPC.Addr,
 		"--wallet", validatorWallet,
 		"--from", validatorAddr,
+		"--force",
```
Comment (AnnaShaleva) [inline cli/candidate_test.go:25]: We need to add this flag to `register` command, currently, it's in the test only.

**Comment by AnnaShaleva** [comment]
And `TestNEP17Transfer`'s regexp for network fee should be updated to a human-readable format.

File: cli/candidate_test.go (hunk 6 above)
Comment (fyrchik) [reply cli/candidate_test.go:25]: This flag is for `multitransfer`, not `register`.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**